src/jdk.compiler/share/classes/com/sun/tools/javac/comp/Attr.java

```diff
@@ -5097,6 +5097,14 @@ public void visitAnnotation(JCAnnotation tree) {
         Assert.error("should be handled in annotate");
     }
 
+    @Override
+    public void visitModifiers(JCModifiers tree) {
```

Comment:
question can't we report this error when the erroneous code is being parsed? adding this method to Attr makes me feel a bit uncomfortable, I can understand if this is the only / best way though

Reply:
Parser will report the error, and will wrap the (presumed) modifiers into an erroneous tree, so the AST is reasonable. But JShell runs javac with `--should-stop=at=FLOW`, so the error in parser will not stop the processing, and when `Attr` will look into the erroneous tree, it will fail.

Reply:
I see, thanks for the clarification